common/src/main/java/com/wynntils/models/containers/BankModel.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.models.containers;
+
+import com.wynntils.core.components.Model;
+import com.wynntils.core.components.Models;
+import com.wynntils.core.persisted.Persisted;
+import com.wynntils.core.persisted.storage.Storage;
+import com.wynntils.mc.event.ScreenClosedEvent;
+import com.wynntils.mc.event.ScreenInitEvent;
+import com.wynntils.models.containers.type.SearchableContainerType;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import net.minecraft.client.gui.screens.inventory.AbstractContainerScreen;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class BankModel extends Model {
+    @Persisted
+    private final Storage<Integer> finalBankPage = new Storage<>(21);
+
+    @Persisted
+    private final Storage<Integer> finalBlockBankPage = new Storage<>(12);
+
+    @Persisted
+    private final Storage<Integer> finalBookshelfPage = new Storage<>(10);
+
+    @Persisted
+    private final Storage<Integer> finalMiscBucketPage = new Storage<>(10);
+
+    @Persisted
+    private final Storage<Map<Integer, String>> customBankPageNames = new Storage<>(new TreeMap<>());
+
+    @Persisted
+    private final Storage<Map<Integer, String>> customBlockBankPageNames = new Storage<>(new TreeMap<>());
+
+    @Persisted
+    private final Storage<Map<Integer, String>> customBookshelfPageNames = new Storage<>(new TreeMap<>());
+
+    @Persisted
+    private final Storage<Map<Integer, String>> customMiscBucketPageNames = new Storage<>(new TreeMap<>());
```

Comment:
Maybe these should be configs.. but we don't support them here yet and I won't hold this feature back for that